src/main/java/org/sagebionetworks/bridge/spring/controllers/EmailController.java

```diff
@@ -0,0 +1,79 @@
+package org.sagebionetworks.bridge.spring.controllers;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.http.HttpStatus.SC_UNAUTHORIZED;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.BridgeServiceException;
+import org.sagebionetworks.bridge.models.accounts.AccountId;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+@CrossOrigin
+@RestController
+public class EmailController extends BaseController {
+    private static final Logger LOG = LoggerFactory.getLogger(EmailController.class);
+    
+    /**
+     * An URL to which a POST can be sent to set the user's email notification preference to "off". Cannot turn email
+     * notifications back on through this endpoint. This cannot be part of the public API, because MailChimp doesn't
+     * deal with non-200 status codes. The token that is submitted is set in the configuration, and must match to allow
+     * this call to succeed. Subject to change without warning or backwards compatibility. 
+     */
+    @RequestMapping(method = {GET, POST}, path="/v3/users/self/unsubscribeEmail", produces="text/plain")
+    public String unsubscribeFromEmail() throws Exception {
+        // We catch and return 200s because MailChimp makes a validation call when configuring the web hook, and if it fails,
+        // MailChimp won't persist the configuration. We could also detect the validation call because it has a different
+        // User-Agent than the real callbacks, "MailChimp.com WebHook Validator" versus "MailChimp.com", and always return 
+        // 200 for that validation call.
+        try {
+            // The servlet getParameter() method will extract the value from a query string or from 
+            // POSTed form data, making this much simpler than in Play.
+            String token = request().getParameter("token");
```

Comment:
Under the old code, we would check both the query params and the post body. Does getParameter() check both query params and post body? If not, is this an intended change? Do we know that it works with MailChimp?

Reply:
Yes, it definitely does (I have tested this manually, and verified it in the documentation). If MailChimp sends the same variables as when we wrote this, then it'll work (the formats are very well specified for query strings and application/x-www-form-urlencoded POST data).